controllers/vault.go

```diff
@@ -0,0 +1,281 @@
+	}
+	for index, key := range vaultInitOutput.UnsealKeysHex {
+		s := "unsealhexkey_" + strconv.Itoa(index)
+		data[s] = []byte(key)
+	}
+	data["keyscount"] = []byte(strconv.Itoa(len(vaultInitOutput.UnsealKeysHex)))
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: vaultSecretName,
+		},
+		Data: data,
+	}
+	secretClient := client.CoreV1().Secrets(operatorNamespace)
+	current, err := secretClient.Get(context.Background(), vaultSecretName, metav1.GetOptions{})
+	if err != nil || current == nil {
+		_, err = secretClient.Create(context.Background(), secret, metav1.CreateOptions{})
+	} else {
+		_, err = secretClient.Update(context.Background(), secret, metav1.UpdateOptions{})
+	}
+	if err != nil {
+		log.Printf("Error creating secret: %s\n", err)
+		return err
+	}
+	log.Printf("Created secret")
+	return nil
+}
+
+func getVaultStructFromSecrets(config *rest.Config, client kubernetes.Interface) (*VaultInitStruct, error) {
+	// If the vault is sealed we take the unseal keys in the k8s secret and use them to unseal the vault
+	secretClient := client.CoreV1().Secrets(operatorNamespace)
+	secret, err := secretClient.Get(context.Background(), "vaultkeys", metav1.GetOptions{})
+	if err != nil || secret == nil {
+		return nil, errors.New(fmt.Errorf("We called vaultUnseal but there were no secrets present: %s", err))
+	}
+	count, err := strconv.Atoi(string(secret.Data["keyscount"]))
+	if err != nil {
+		return nil, errors.New(fmt.Errorf("Converting keys count failed: %s", err))
+	}
+	var v VaultInitStruct
+	v.RootToken = string(secret.Data["roottoken"])
+	v.UnsealKeysHex = []string{}
+	for i := 0; i < count; i++ {
+		v.UnsealKeysHex = append(v.UnsealKeysHex, string(secret.Data["unsealhexkey_"+strconv.Itoa(i)]))
+	}
+	err = unsealVaultOperator(config, client, &v)
+	if err != nil {
+		return nil, err
+	}
+	return &v, nil
+}
+
+func unsealVaultOperator(config *rest.Config, client kubernetes.Interface, vaultInitOutput *VaultInitStruct) error {
+	var errCount int = 0
+	if len(vaultInitOutput.UnsealKeysHex) == 0 || len(vaultInitOutput.UnsealKeysHex) < vaultInitOutput.UnsealThreshold {
+		return errors.New("We do not have sufficient keys to unseal the vault")
+	}
+
+	for _, key := range vaultInitOutput.UnsealKeysHex {
+		_, _, err := execInPod(config, client, "vault", "vault-0", "vault", []string{"vault", "operator", "unseal", key})
+		if err != nil {
+			errCount += 1
+			log.Printf("Error while processing %s: -> %s", key, err)
+		}
+	}
+	if errCount > 0 {
+		return errors.New("Errored while calling vault operator unseal")
+	}
+
+	log.Printf("Vault successfully unsealed")
+	return nil
+}
+
+func vaultStatus(config *rest.Config, client kubernetes.Interface) (*VaultStatus, error) {
+	if !haveNamespace(client, "vault") {
+		return nil, errors.New(fmt.Errorf("'vault' namespace not found yet"))
+	}
+	if !havePod(client, "vault", "vault-0") {
+		return nil, errors.New(fmt.Errorf("'vault/vault-0' pod not found yet"))
```

Comment:
```suggestion
		return nil, fmt.Errorf("'vault/vault-0' pod not found yet")
```

I don't think we need the `errors.New()` right?

Reply:
You're right it should not be needed